### PR TITLE
Various styling adjustments for new nav

### PR DESCRIPTION
**Pull request by mmmavis**
@danielfmiranda This PR contains quite a big of styling fixes and can be tricky to do in-depth code review. However, I will say since these are all styling related fixes, as long as we get the okay from designers and the code changes don't look concerning to you, it should be safe to merge. 😁 

---

### Changes
- showed desktop styled nav on xlarge screens instead of large
- fixed various spacing issues
- changed 'title' to 'aria-label' to avoid tooltip

# Description

Link to sample test page: https://foundation-s-12289-nav--ns14nh.herokuapp.com/ (CMS creds: `admin2/admin2password`)
Related PRs/issues: #12289 

┆Issue is synchronized with this [Jira Story](https://mozilla-hub.atlassian.net/browse/TP1-613)

### CODE REVIEW

File: tailwind-plugins/site-nav.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+const plugin = require("tailwindcss/plugin");
```
Comment (mmmavis) [inline tailwind-plugins/site-nav.js:1]: I'm creating this `site-nav.js` Tailwind plugin in this PR so we can create new utility class for the nav work. Once we can fully sunset the old nav, my plan is to clean up the template code a bit by converting the loooong class lists into a new class defined in this file.

**Review by nancyt1 — COMMENTED** [review]
Not sure if it's just the review app but when I open a section, I can't scroll all the way down to the newsletter button and the newsletter button doesn't work. But all other spacing looks good!

One mobile change:
SCR-20240506-oapb

**Comment by mmmavis** [comment]
@nancyt1 Thanks for the feedback. It seems like the newsletter button functionality is also broken on staging/production. I will file a new ticket for it. 

> Not sure if it's just the review app but when I open a section, I can't scroll all the way down to the newsletter button 

This seems to be regression introduced in this PR. I'll take a look. 

> Spacing

SCR-20240506-oapb

Did you mean increasing all sections' bottom padding by 1rem?

**Comment by nancyt1** [comment]
@mmmavis If there isn't padding already at the bottom of the sections then yes. But it seemed like all other sections with a button or had the featured section had padding except ones that end in just a nav column.

**Comment by mmmavis** [comment]
@nancyt1 I adjusted the bottom padding as noted in your screencap to 40px/2.5rem. Let me know what you think!

For the two issues related to newsletter button - I realized they are both appearing on staging and not newly introduced in this PR. I have created a new ticket and the devs will tackle in a different PR.

**Review by nancyt1 — APPROVED** [review]
Lovely! Thank you!

**Comment by mmmavis** [comment]
I managed to find a fix to make newsletter button show again. Gonna slide the fix into this PR.

To code reviewer / @danielfmiranda -
Here's the [commit](https://github.com/MozillaFoundation/foundation.mozilla.org/pull/12296/commits/0c1e28cbd79f7c6ca4d548d362804050efa7a1f8). Since primary_nav.html is shared in between the old nav and new nav (and MozFest site), I had to find a new way to target `.narrow-screen-menu` on the new nav. I've done so by adding a new class `new-nev-enabled` to `<body>` and move some styling over to SCSS so they can be specific for the new nav. This is not the cleanest solution but is the safest as we have too many code shared by different variation of the navs. I can wait for the day to tidy up the code!

File: source/sass/components/primary-nav.scss [hunk 2]
```diff
@@ -245,11 +245,12 @@
...
       overflow-y: auto;
-      height: 100vh;
-      transition: all 0.5s;
+      transition:
+          opacity 0.2s,
+          height 0.5s;
 
       &.hidden {
-        height: 0;
+        opacity: 0;
```
Comment (mmmavis) [inline source/sass/components/primary-nav.scss:253]: adding this rule back as the old nav needs it

**Review by danielfmiranda — APPROVED** [review]
Looks great! Thanks @mmmavis 👍 approved!

**Comment by data-sync-user** [comment]
➤ Simon Acosta Torres commented:

PR has already been merged.

**Pull request merged**